lib/models/service.ts

```diff
@@ -0,0 +1,291 @@
+/*
+Copyright 2018 Balena
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as errors from 'balena-errors';
+import * as Promise from 'bluebird';
+import once = require('lodash/once');
+import {
+	PineOptionsFor,
+	Service,
+	ServiceEnvironmentVariable,
+} from '../../typings/balena-sdk';
+import { InjectedDependenciesParam, InjectedOptionsParam } from '../balena';
+import { findCallback, mergePineOptions } from '../util';
+
+const getServiceModel = (
+	deps: InjectedDependenciesParam,
+	opts: InjectedOptionsParam,
+) => {
+	const { pine } = deps;
+	const applicationModel = once(() => require('./application')(deps, opts));
+	const { buildDependentResource } = require('../util/dependent-resource');
+
+	const varModel = buildDependentResource(
+		{ pine },
+		{
+			resourceName: 'service_environment_variable',
+			resourceKeyField: 'name',
+			parentResourceName: 'service',
+			getResourceId: (id: number) => get(id, { $select: 'id' }).get('id'),
+			ResourceNotFoundError: errors.BalenaServiceNotFound,
+		},
+	);
+
+	exports = {};
+
+	// Not exported for now, but we could document & export in future
+	// if there are external use cases for this.
+	const get = (id: number, options: PineOptionsFor<Service>) => {
```

Comment:
How about:
```suggestion
	const get = (id: number, options: PineOptionsFor<Service> = {}) => {
```
Applies in the rest of the file as well.

Reply:
I just used decaffeinate and didn't really think of doing this